cli/src/cast.rs

```diff
@@ -683,6 +690,56 @@ async fn main() -> eyre::Result<()> {
             let selector = HumanReadableParser::parse_function(&sig)?.short_signature();
             println!("0x{}", hex::encode(selector));
         }
+        Subcommands::Create2 { starts_with, ends_with, deployer, init_code } => {
+            let mut regexs = vec![]; // TODO: add --matching param
+            if let Some(prefix) = starts_with {
+                let pad_width = prefix.len() + prefix.len() % 2;
+                hex::decode(format!("{:0>width$}", prefix, width = pad_width))
+                    .expect("invalid prefix hex provided");
+                regexs.push(format!(r"^{}", prefix));
+            }
+            if let Some(suffix) = ends_with {
+                let pad_width = suffix.len() + suffix.len() % 2;
+                hex::decode(format!("{:0>width$}", suffix, width = pad_width))
+                    .expect("invalid suffix hex provided");
+                regexs.push(format!(r"{}$", suffix));
+            }
+
+            assert!(
+                regexs.iter().map(|p| p.len() - 1).sum::<usize>() <= 40,
+                "vanity patterns length exceeded. cannot be more than 40 characters",
+            );
+
+            let regex = RegexSet::new(regexs)?;
+
+            let init_code = Bytes::from(init_code.into_bytes());
+
+            println!("Starting to generate deterministic contract address...");
+            let timer = Instant::now();
+            let salt = std::iter::repeat_with(move || H256::random_using(&mut thread_rng()))
+                .par_bridge()
+                .find_any(|salt| {
+                    let salt = Bytes::from(salt.to_fixed_bytes());
+                    let init_code = init_code.clone();
+
+                    let addr = get_create2_address_from_hash(deployer, salt, init_code);
+                    let addr = hex::encode(addr);
+
+                    regex.matches(&addr).into_iter().count() == regex.patterns().len()
```

Comment:
do we need a regex here?
can we just compare if hex bytes starts/ends with?

Reply:
OP wanted to use `matches` so I think that I will remove this, is it bad performance wise to use regex rather than comparing each character or a slice of the sequence ?
Also, this regex is copy pasted from the Vanity addresses code.

Reply:
so no more `starts_with` / `ends_with` arguments?

how would `matches` behave?

imo `starts_with` / `ends_with` would be more explicit and probably what the user is interested in, an address that starts, and/or ends with a certain pattern


oh so `starts_with` and `ends_with` should support Regex? then this makes sense

Reply:
Still `start`/`end_with` but the `matching` will be added as well, but they cannot be together because that doesn't makes sense.
I was just wondering, if matching is added, I don't think that we will need `start`/`end_with` in the solidity library that will use `ffi` because matching is more explicit and less handy to type on the command line.
Anyway, maybe that's not a big deal because the `mineSalt` (I think) will be external to foundry, right ?
And even with that, it should not be a big pain to add a function that overloads the ones that uses `matching`